core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +182,24 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
```

Comment:
can we add java docs about the behavior and relationship between index level ack and shard level ack?